### PR TITLE
chore: update webpack loaders

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request mainly takes a shot at updating the webpack loaders which we failed to update in the past. In all to be able to more easily update to the next version of webpack.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,7 +107,7 @@
     "lint-staged": "8.2.1",
     "mri": "1.1.4",
     "postcss": "7.0.17",
-    "postcss-custom-properties": "8.0.11",
+    "postcss-custom-properties": "9.0.1",
```
Comment (tdeekens) [inline package.json]: It mainly updated an internal dep and dropped node@6 support.

File: packages/mc-scripts/config/create-webpack-config-for-development.js [hunk 2]
```diff
@@ -215,9 +214,12 @@ module.exports = ({
           {
             loader: require.resolve('css-loader'),
             options: {
-              modules: true,
+              modules: {
+                mode: 'local',
```
Comment (tdeekens) [inline packages/mc-scripts/config/create-webpack-config-for-development.js]: This is the new way of configuring it. css-modules is disabled by default now and the modules is a boolean or object.

File: packages/mc-scripts/config/create-webpack-config-for-development.js [hunk 3]
```diff
@@ -215,9 +214,12 @@ module.exports = ({
...
+                localIdentName: '[name]__[local]___[hash:base64:5]',
+                hashPrefix: 'ct',
```
Comment (tdeekens) [inline packages/mc-scripts/config/create-webpack-config-for-development.js]: Not sure. Just thought that would be nice.

File: packages/mc-scripts/config/create-webpack-config-for-production.js [hunk 4]
```diff
@@ -160,8 +160,10 @@ module.exports = ({
     },
 
     plugins: [
-      new CleanWebpackPlugin([distPath], {
-        allowExternal: true,
+      new CleanWebpackPlugin({
+        dangerouslyAllowCleanPatternsOutsideProject: true,
+        dry: true,
+        cleanOnceBeforeBuildPatterns: [distPath],
```
Comment (tdeekens) [inline packages/mc-scripts/config/create-webpack-config-for-production.js]: This now hooks into the webpack config and normally doesn't need the dist path. However in this case we do need it as we seem to clean outside of the project.

File: packages/mc-scripts/package.json [hunk 5]
```diff
@@ -70,11 +70,11 @@
     "regenerator-runtime": "0.13.2",
     "shelljs": "0.8.3",
     "style-loader": "0.23.1",
-    "svg-url-loader": "2.3.3",
+    "svg-url-loader": "3.0.0",
     "terser-webpack-plugin": "1.3.0",
     "thread-loader": "2.1.2",
-    "url-loader": "1.1.2",
-    "webpack": "4.35.0",
+    "url-loader": "2.0.1",
```
Comment (tdeekens) [inline packages/mc-scripts/package.json]: Literally changes I don't understand and I dare to assume we don't rely on. At least I've never seen those things mentioned in the changelog.

File: package.json (hunk 1 above)
Comment (montezume) [reply package.json]: Be careful, we had problems with this update

Comment (montezume) [reply package.json]: I had a lot of problems with this version 

https://github.com/postcss/postcss-custom-properties/issues/186

Comment (tdeekens) [reply package.json]: Okay so best to not update this one I presume.

File: packages/mc-scripts/config/create-webpack-config-for-development.js (hunk 3 above)
Comment (tdeekens) [reply packages/mc-scripts/config/create-webpack-config-for-development.js]: This is confusing. The `css-loader` states that "options should NOT have additional properties" which should come from a validation error against their schema [here](https://github.com/webpack-contrib/css-loader/blob/master/src/options.json). However, I think we don't violate this schema. There are also issues people have run into with this on GitHub. Also when changing it to `modules: true` the same error comes around.

**Comment by montezume** [comment]
Do you need help with this @tdeekens ?

**Comment by tdeekens** [comment]
If you care to look into it or have a clue. I am a bit clueless why the loader rejects a valid config.

**Pull request merged**